File: December - 12/SakthiDharan01_missing_positive_element.py
import sys

def solve():
    data = list(map(int, sys.stdin.read().strip().split()))
    if not data:
        return
    n = data[0]
    arr = data[1:1+n]
    expected_sum = n * (n + 1) // 2
    expected_sq = n * (n + 1) * (2 * n + 1) // 6
    actual_sum = sum(arr)
    actual_sq = sum(x * x for x in arr)
    diff = expected_sum - actual_sum  # missing - duplicate
    diff_sq = expected_sq - actual_sq  # missing^2 - duplicate^2 = (m-d)(m+d)
    sum_md = diff_sq // diff  # missing + duplicate
    missing = (diff + sum_md) // 2
    duplicate = missing - diff
    print(f"Missing Number: {missing}")
    print(f"Duplicate Number: {duplicate}")

if __name__ == "__main__":
    solve()

File: December - 12/SakthiDharan01_missing_positive_element.java
import java.util.*;

public class SakthiDharan01_missing_positive_element {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt()) return;
        int n = sc.nextInt();
        long expectedSum = (long) n * (n + 1) / 2;
        long expectedSq = (long) n * (n + 1) * (2L * n + 1) / 6;
        long actualSum = 0;
        long actualSq = 0;
        for (int i = 0; i < n && sc.hasNextInt(); i++) {
            long v = sc.nextLong();
            actualSum += v;
            actualSq += v * v;
        }
        long diff = expectedSum - actualSum; // missing - duplicate
        long diffSq = expectedSq - actualSq; // (m-d)(m+d)
        long sum = diffSq / diff; // missing + duplicate
        long missing = (diff + sum) / 2;
        long duplicate = missing - diff;
        System.out.println("Missing Number: " + missing);
        System.out.println("Duplicate Number: " + duplicate);
        sc.close();
    }
}
